### PR TITLE
Add tests for plot and movie and QG

### PR DESCRIPTION
**Pull request by AndreyAPopov**

### CODE REVIEW

**Comment by Steven-Roberts** [comment]
I'm ok with merging this (maybe after #95). I haven't reviewed carefully, but I'll go though all the tests together carefully in the future

**Comment by elswit** [comment]
Added  fix for #91

File: tests/+problemtests/validateqg.m [hunk 1]
```diff
@@ -0,0 +1,39 @@
+function validateqg
+
+fprintf(' Testing Quasi-Geostrophic Equations\n');
+
+try
+    model = otp.quasigeostrophic.presets.PopovMouIliescuSandu('size', [16, 32]);
+    model.TimeSpan = [0, 0.109];
+catch
+    % if it failed to build, other tests would show it.
```
Comment (Steven-Roberts) [inline tests/+problemtests/validateqg.m:9]: I think the try catch is unnecessary for this reason

File: tests/validateallderivatives.m [hunk 2]
```diff
@@ -19,15 +19,15 @@
         presetname);
 
     try
-        model = eval(presetclass);
+        problem = eval(presetclass);
     catch
         continue;
```
Comment (Steven-Roberts) [inline tests/validateallderivatives.m:24]: Probably don't need try catch

File: tests/validateallplots.m [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    presetclass = preset.presetclass;
+
+    fprintf(' %-20s | %-20s | ', ...
+        problemname, ...
+        presetname);
+
+    try
+        problem = eval(presetclass);
+    catch
+        continue;
```
Comment (Steven-Roberts) [inline tests/validateallplots.m:24]: Same as above

File: tests/validateallplots.m [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    try
+        problem.plot(sol);
+        fprintf('PASS | ');
+    catch e
+        assert(false, sprintf('Preset %s of %s failed to plot with error\n%s\n%s', ...
+            presetname, problemname, e.identifier, e.message))
+        fprintf('-FAIL ');
+    end
+
+    if problem.NumVars > 1
```
Comment (Steven-Roberts) [inline tests/validateallplots.m:67]: It still works with 1 var

File: tests/validateallplots.m [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            problem.TimeSpan = [0, 0.01];
+        end
+        sol = problem.solve();
+        assert(true)
+    catch
+        continue;
+    end
+
+    try
+        problem.plot(sol);
```
Comment (Steven-Roberts) [inline tests/validateallplots.m:59]:
```
fig = problem.plot(sol);
close(fig)
```

**Review by Steven-Roberts — CHANGES_REQUESTED** [review]

File: tests/validateallplots.m (hunk 4 above)
Comment (AndreyAPopov) [reply tests/validateallplots.m:67]: It doesn't work for me.

**Comment by Steven-Roberts** [comment]
Add option to test plots but not movies in Octave. We can test what will happen on GitHub by running Octave without graphics

**Comment by Steven-Roberts** [comment]
TODO: fix formatting of PASS in table for some problems

**Comment by AndreyAPopov** [comment]
I think this one is ready

**Pull request merged**